src/dtx/dtx_internal.h

```diff
@@ -56,6 +58,21 @@ enum dtx_operation {
 
 CRT_RPC_DECLARE(dtx, DAOS_ISEQ_DTX, DAOS_OSEQ_DTX);
 
+/* DTX collective RPC input fields */
```

Comment:
The feature is "collectively punch" instead of punch object with collective RPC. The "collective" in the comment means the local collective operation on each engine.

As for "DAOS_ISEQ_REMOTE_DTX or DAOS_ISEQ_PARTICIPANT_DTX", the current "DTX" already contains the content of both local and remote operations, they all be part of participants for the DTX. So these two names are more confused.

As for using collective RPC, I would say yes, that will be helpful when forwarding a lot of IO RPCs from leader to non-leaders. But such part on server are general framework for the DTX logic for all update/punch/CPD operations. It is not easy to simply replace it with bcast. The main trouble is that current collective RPC does not support to send different RPCs to different engines. I hope we can enable/change related logic in another independent ticket. Since that are only server side changes, it will not cause interoperability issue even if we change in the future.